src/deployment/deploylib/registration.py

```diff
@@ -270,19 +270,54 @@ def create_application_registration(
         "appId": registered_app["appId"],
     }
 
-    query_microsoft_graph(
-        method="POST",
-        resource="servicePrincipals",
-        body=service_principal_params,
-        subscription=subscription_id,
-    )
+    def try_sp_create() -> None:
+        error: Optional[Exception] = None
+        for _ in range(10):
+            try:
+                query_microsoft_graph(
+                    method="POST",
+                    resource="servicePrincipals",
+                    body=service_principal_params,
+                    subscription=subscription_id,
+                )
+                return
+            except GraphQueryError as err:
+                # work around timing issue when creating service principal
+                # https://github.com/Azure/azure-cli/issues/14767
+                if (
+                    "service principal being created must in the local tenant"
+                    not in str(err)
+                ):
+                    raise err
+            logger.warning(
+                "creating service principal failed with an error that occurs "
+                "due to AAD race conditions"
+            )
+            time.sleep(60)
+        if error is None:
+            raise Exception("service principal creation failed")
+        else:
+            raise error
+
+    try_sp_create()
+
+    registered_app_id = registered_app["appId"]
+    app_id = app["appId"]
+
+    def try_authorize_application(data: Any) -> None:
+        authorize_application(
+            UUID(registered_app_id),
+            UUID(app_id),
+            subscription_id=subscription_id,
+        )
+
+    retry(try_authorize_application, "authoriz application")
```

Comment:
```suggestion
    retry(try_authorize_application, "authorize application")
```

Reply:
Yup I'm tracking that. Just retriggering builds.